apps/api/src/database/repositories/competition-repository.ts

```diff
@@ -725,6 +727,69 @@ export async function getAgentPortfolioSnapshots(
   }
 }
 
+/**
+ * Get the newest and oldest portfolio snapshots for an agent in a competition
+ * @param competitionId Competition ID
+ * @param agentId Agent ID
+ * @returns Object with newest and oldest snapshots, or null if no snapshots found
+ */
+export async function getBoundedSnapshots(
```

Comment:
I built this function to help reduce load when we want to calculate the stats for an agent when a competition is ended.
There were places we were using `getAgentPortfolioSnapshots` for this, and that returns all the snapshots for a comp+ agent combo, but we only need the first and last.